src/intersections/segments.jl

```diff
@@ -79,7 +79,9 @@ function intersecttype(s1::Segment{2,T}, s2::Segment{2,T}) where {T}
       NonIntersectingSegments()
     end
   elseif !determinatex && !determinatey # CASE (III)
-    if !isapprox((x2 - x1) × (y2 - y1), zero(T), atol=atol(T))
+    u, v = (x2 - x1), (y2 - y1)
+    α = (u ⋅ v) / (v ⋅ v)
+    if !isapprox(norm(u - α*v), zero(T), atol=atol(T))
```

Comment:
```suggestion
    if !isapprox(u × v, zero(T), atol=atol(T)^2)
```
It passes tests on my machine (including the additional one from the PR), and is way simpler. Would there be any reason for not using `atol(T)^2`?

Reply:
Also, now that `u` and `v` are not used multiple times, it may be better to just leave this as it was, with the only change being that we test for `atol(T)^2`, not introducing `u` and `v`.

Reply:
Nice. I was not sure if the `atol(T)^2` was safe. Perhaps there is a material somewhere explaining how these floating point errors propagate in area calculations? I will commit your suggestion and we see if we bump into this again in the future. 👍🏽

Reply:
The beginning of [this video](https://www.youtube.com/watch?v=zHPXGBiTM5A) explains very well how floats behave in numerical operations. If I understood correctly, when comparing to `zero(T)`, it's just going to check whether the result is between `[-atol(T), atol(T)]` since `zero(T)` is represented perfectly on floats, so there should be no problem here. Problems would arise if, for example, you do an area calculation, then take the square root of that.